Nimble/Matchers/AllPass.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+
+public func allPass<T,U where U: SequenceType, U.Generator.Element == T>
+    (passFunc: (T?) -> Bool) -> NonNilMatcherFunc<U> {
+        return allPass("pass a condition", passFunc)
+}
+
+public func allPass<T,U where U: SequenceType, U.Generator.Element == T>
+    (passName:String, passFunc: (T?) -> Bool) -> NonNilMatcherFunc<U> {
+        return createAllPassMatcher() {
+            expression, failureMessage in
+            failureMessage.postfixMessage = passName
+            return passFunc(expression.evaluate())
+        }
+}
+
+public func allPass<U,V where U: SequenceType, V: NonNilBasicMatcher, U.Generator.Element == V.ValueType>
+    (matcher: V) -> NonNilMatcherFunc<U> {
+        let wrapper = NonNilMatcherWrapper(NonNilBasicMatcherWrapper(matcher))
+        return createAllPassMatcher() {wrapper.matches($0, failureMessage: $1)}
+}
+
+public func allPass<U,V where U: SequenceType, V: BasicMatcher, U.Generator.Element == V.ValueType>
+    (matcher: V) -> NonNilMatcherFunc<U> {
+        let wrapper = BasicMatcherWrapper(matcher: matcher)
+        return createAllPassMatcher() {wrapper.matches($0, failureMessage: $1)}
+}
+
+public func allPass<U,V where U: SequenceType, V: Matcher, U.Generator.Element == V.ValueType>
```

Comment:
This is tangential to this pull request, but I recently created a matcher similar to this one: https://github.com/modocache/Guanaco/blob/18126c0c8ba9a7cb3cc5d43aec83c81047410ab6/Guanaco/HaveSucceeded.swift#L21-L49

I also had to write several matcher functions in order to handle `Matcher`, `BasicMatcher`, and `NonNilBasicMatcher`. Architecturally, this seems like a shortcoming. Any thoughts on improvements here, @jeffh?